File: frontend/src/types/Asset.ts
export type AssetItem = {
    id: string;
    assetName: string;
    category: string;
    status: string;
    managementNumber: string;
    purchaseDate: string | null;
    remarks: string;
};

export type AssetSearchCondition = {
    assetName: string;
    category: string;
    status: string;
}

export type AssetLayoutContext = {
    assets: AssetItem[];

    setAssets: React.Dispatch<
        React.SetStateAction<AssetItem[]>
    >;

    searchCondition: AssetSearchCondition;

    setSearchCondition: React.Dispatch<
        React.SetStateAction<AssetSearchCondition>
    >;

    appliedSearchCondition: AssetSearchCondition;

    setAppliedSearchCondition: React.Dispatch<
        React.SetStateAction<AssetSearchCondition>
    >;
};

File: backend/AssetManagementSystem.Api/Entities/Asset.cs
namespace AssetManagementSystem.Api.Entities
{
    public class Asset
    {
        public Guid Id { get; set; }

        public string AssetName { get; set; } = string.Empty;

        public string Category { get; set; } = string.Empty;

        public string Status { get; set; } = string.Empty;

        public string ManagementNumber { get; set; } = string.Empty;

        public DateOnly? PurchaseDate { get; set; }

        public string Remarks { get; set; } = string.Empty;
    }
}
